packages/dds/tree/src/test/utils.ts

```diff
@@ -511,10 +515,11 @@ export class SharedTreeTestFactory extends SharedTreeFactory {
 	 * @param onLoad - Called once for each tree that is loaded from a summary.
 	 */
 	public constructor(
-		private readonly onCreate: (tree: SharedTree) => void,
-		private readonly onLoad?: (tree: SharedTree) => void,
+		protected readonly onCreate: (tree: SharedTree) => void,
+		protected readonly onLoad?: (tree: SharedTree) => void,
+		options: SharedTreeOptionsInternal = { jsonValidator: typeboxValidator },
 	) {
-		super({ jsonValidator: typeboxValidator });
+		super(options);
```

Comment:
Is this better?

```suggestion
		super({ ...options, jsonValidator: typeboxValidator });
```

You want this to use `typeboxValidator` no matter what, correct? Otherwise it's broken? Or, is there a legitimate use case for a subclass of this which allows a different validator?

And if so, you'd still want to supply `typeboxValidator` if the subclass _didn't_ supply anything. So that'd at least be:

`super({ ...options, jsonValidator: options.jsonValidator ?? typeboxValidator })`

In either case, you can change the default value for the `options` parameter to be `{}`, and you should follow the same pattern in the other constructor of your new subclass (because _it_ could be subclassed).

Reply:
Updated with suggestion